### DISCUSSION TITLE
Define DG discontinuity across an internal boundary

### DISCUSSION THREAD

**Discussion by Heinrich-BR** [mfem/mfem#3850] [Q&A]
Hi, I'm currently trying to use a DG integrator on a problem that has an internal boundary. I would like, if possible, to tell the integrator that the quantity that I'm solving for has a discontinuity across this internal boundary. For example, imagine I want to solve a diffusion problem in a domain but I have a boundary with temperature T1 on one side and T2 on the other side. Is this possible to do? Thank you for the help!

**Comment by vladotomov** [comment]
Yes, you can use a `Coefficient` that gives the jump in temperature.

**Reply by Heinrich-BR** [reply]
Thank you @vladotomov! How exactly can I go about doing this? As far as I know, `Coefficient` only allows you to define a value, not a discontinuity.

**Reply by jandrej** [reply]
Your Coefficient can vary spatially. E.g. a `FunctionCoefficient` that returns 
```
if (x > 1.0) {
 return 150.0;
} else if (x <= 1.0) {
 return -100.0;
}
```

**Reply by Heinrich-BR** [reply]
Ah right, I see what you mean. That could work in any specific case where you know where the boundary is (in your example the boundary is at x=1.0), but is it possible to generalise it? Like to define the `FunctionCoefficient` not with respect to global coordinates, but locally with respect to a certain boundary that can be in any direction and even curved?

**Reply by termi-official** [reply]
I usually introduce the discontinuities via the element attribute, where my mesh is generated in such a way that the boundary is represented geometrically. Note that in the eval call to coefficients there is the `ElementTransformation` passed which carries the attribute information (https://docs.mfem.org/html/classmfem_1_1ElementTransformation.html#adcbe93689e9f88e401464c49f0d4fb2e). Alternatively the `ElementTransformation` also carries the element id.

If there is a non-trivial coupling between the domains, then you also might want to consider to actually model the subdomains with the `SubMesh`, where the interface is handled manually.
